catalog/spatial/registry/registry-admin-modules/registry-admin-local-ui/src/main/webapp/js/model/Segment.js

```diff
@@ -0,0 +1,464 @@
+            }
+            return slot.value[0];
+        } else {
+            if (type === 'boolean') {
+                return slot.value === 'true' ? true : false;
+            } else if (type === 'point') {
+                return {coords: slot.value.Point.pos.split(/[ ]+/)};
+            }
+            else if (type === 'bounds') {
+                var lowerCorner = slot.value.Envelope.lowerCorner.split(/[ ]+/);
+                var upperCorner = slot.value.Envelope.upperCorner.split(/[ ]+/);
+                return {coords: upperCorner.concat(lowerCorner)};
+            } else {
+                return slot.value;
+            }
+        }
+    }
+
+    function generateId() {
+        counter++;
+        return 'urn:segment:id:' + new Date().getTime() + "-" + counter;
+
+    }
+
+    function getSegment(data, key) {
+        var dataModel = data;
+        if (!_.isArray(data)) {
+            dataModel = [data];
+        }
+        for (var index = 0; index < dataModel.length; index++) {
+            var foundModel;
+
+            if (dataModel[index].id === key) {
+                return dataModel[index];
+            }
+
+            for (var prop in dataModel[index]) {
+                if (dataModel[index].hasOwnProperty(prop) && typeof dataModel[index][prop] === 'object') {
+                    foundModel = getSegment(dataModel[index][prop], key);
+                    if (foundModel) {
+                        return foundModel;
+                    }
+                }
+            }
+        }
+    }
+
+    var Segment = {};
+
+    Segment.Segment = Backbone.AssociatedModel.extend({
+        relations: [
+            {
+                type: Backbone.Many,
+                key: 'segments',
+                collectionType: function () {
+                    return Segment.Segments;
+                },
+                relatedModel: function () {
+                    return Segment.Segment;
+                }
+            },
+            {
+                type: Backbone.Many,
+                key: 'fields',
+                collectionType: function () {
+                    return Field.FormFields;
+                },
+                relatedModel: function () {
+                    return Field.FormField;
+                }
+            }
+        ],
+        defaults: {
+            parentId: undefined,
+            simpleId: undefined,
+            segmentId: undefined,
+            segmentName: undefined,
+            segmentType: undefined,
+            containerOnly: false,
+            multiValued: false,
+            nestedLevel: 0,
+            fields: [],
+            segments: [],
+            associationModel: undefined
+        },
+        populateFromModel: function (dataModel, descriptors) {
+            var model = this;
+            var segs = [];
+            var seg;
+            var segType = model.get('segmentType');
+            model.dataModel = dataModel;
```

Comment:
Something doesn't seem quite right with this. What is this dataModel variable that is attached to the model? Generally speaking, if you're attaching something directly to the backbone model itself, you're doing it wrong.